get_http.go

```diff
@@ -117,13 +118,54 @@ func (g *HttpGetter) GetFile(dst string, u *url.URL) error {
 		g.Client = httpClient
 	}
 
-	resp, err := g.Client.Get(u.String())
+	// check to see whether user has specified a range of bytes to download.
+	// if user has, but the range is invalid, fall back to downloading the
+	// whole file
+	byteRange, rangeErr := GetByteRange(u)
+	if rangeErr != nil {
+		if strings.Compare("No byte range provided", rangeErr.Error()) != 0 {
+			fmt.Printf("%s; going to download entire file without a range request",
+				rangeErr)
+		}
+	}
```

Comment:
I'd remove the entire if statement above and invert the logging logic: only log when we _do_ specify a range request. That is the far more unlikely scenario (most HTTP requests do not have range requests) so it will result in a less noisy log I believe.

Also for logging, I think we should be using `log` here. I don't _think_ this package is updated to `go-hclog`, so its probably `log`. I'd just grep the package and see what the style is... I can't remember off the top of my head sorry.

Reply:
This is unclear and I'll try to clean it up so it reads better but the current logic is that it only logs if the user provided a byte range but the code was unable to parse it, resulting in the range request being ignore and the entire file being downloaded.  I can log this as a warning instead though.

I'll replace this with the log package.

Reply:
Ah sorry, it was late so unsurprising I made a mistake! In that case with the other suggestions in this PR I'd say we do a basic `if err != nil` and use `err == nil && result == nil` to specify no range request.

Reply:
Sounds good.  I'll noodle on this today.